department-of-veterans-affairs/caseflow
# Program Office & VISN users can mark a task as "In Progress"

Resolves [CASEFLOW-2283](https://vajira.max.gov/browse/CASEFLOW-2283)

### Description
These changes give VHA Program Office & VISN users a way to indicate to other users in their offices that they are actively working on an appeal.

### Acceptance Criteria
- [ ] Program Office users have an available "Mark task in Progress" task action
- [ ] VISN users have an available "Mark task in Progress" task action
- [ ] Users are redirected to their queue "In Progress" tab and presented with a success message telling them that the status of the task is "In Progress"
- [ ] Cases in a Program Office queue and VISN queues that are marked as "In Progress" in case details appear in the "In Progress" tab in respective queues
- [ ] Once a user has marked a task "In Progress", no other user can mark the task as in progress.

### Testing Plan
1. Sign in as a BVA Intake user and complete the intake of a 10182 with a VHA Benefit Type which will subsequently appear in the CAMO organization queue
2. Sign in as a CAMO user and assign out the appeal to any Program Office
3. Sign in as a Program Office user and visit the organization queue of the respective office you've assigned the appeal to (Caregiver etc.) and select the appeal
4. Note the `uuid` of the appeal so you can examine the task tree in your console
5. Select the "Mark task in progress " action and submit
6. Make sure you see the correct success message banner as seen in the designs when redirected back to the queue
7. Check the task tree of the appeal in the console and ensure that the `AssessDocumentationTask` has a status of `in_progress`
8. Revisit the appeal as the PO and after confirming that you no longer have the "Mark task in progress" as an available action, select the "Assign to VISN" task action
9. Sign in as a VISN user and visit the VISN queue of the org you assigned the appeal to and select the appeal
10. Repeat steps 5 - 7 while also confirming the "Mark task in progress" is no longer available to confirm same functionality for VISN

### User Facing Changes

 BEFORE|AFTER
 ---|---
![Screen Shot 2021-11-03 at 5 53 32 PM](https://user-images.githubusercontent.com/18618189/140198428-8ccbb60d-ccf1-401b-b62a-16f4e385d33d.png) | ![Screen Shot 2021-11-03 at 5 53 49 PM](https://user-images.githubusercontent.com/18618189/140198445-4c6960ae-d47a-4f2d-a4e2-eb73dd5881d9.png)
